Remove unused exception parameter from some files, to work with -Wunused-exception-parameter

psiddh opened the pull request:
Summary: address compiler complains that were coming up to unblock the build

Test Plan:
before the change
```
aten/src/ATen/native/LinearAlgebra.cpp:3623:36: error: unused exception parameter 'e' [-Werror,-Wunused-exception-parameter]
 3623 |     } catch (const std::exception& e) {
      |
```

after: targets build with `-Wunused-exception-parameter`

Differential Revision: D84876246

seemethere commented:
/easycla

Skylion007 reviewed (approved):
Huh, because TORCH_WARN might be removed based on various compile switches, interesting...

cyyever reviewed (approved):
Hope that we can enable Wunused-exception-parameter finally.

tonytargonski committed:
Remove unused exception parameter from some files, to work with -Wunused-exception-parameter (#165770)

Summary:

address compiler complains that were coming up to unblock the build

Test Plan:
before the change
```
aten/src/ATen/native/LinearAlgebra.cpp:3623:36: error: unused exception parameter 'e' [-Werror,-Wunused-exception-parameter]
 3623 |     } catch (const std::exception& e) {
      |
```

after: targets build with `-Wunused-exception-parameter`

Differential Revision: D84876246

clee2000 commented:
@pytorchbot merge -f "reran the job and it succeeded"
